sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -858,6 +859,30 @@ object PushPredicateThroughJoin extends Rule[LogicalPlan] with PredicateHelper {
 }
 
 /**
+  * Push [[Limit]] operators through [[Join]] operators, iff the join type is outer joins.
+  * Adding extra [[Limit]] operators on top of the outer-side child/children.
+  */
+object PushLimitThroughOuterJoin extends Rule[LogicalPlan] with PredicateHelper {
```

Comment:
For example, we have the a table A having 1, 2, 3, 4, 5 (say k is the column name). If we do `A x FULL OUTER JOIN A y ON (x.k = y.k) limit 2` and we push limit to both side, it is possible that we get `1,2` from the left side and `3, 4` from the right side, right?